test-data/unit/stubgen.test

```diff
@@ -741,6 +769,8 @@ x = 1
 class C:
     def g(self): ...
 [out]
+__all__ = ['f', 'x', 'C']
```

Comment:
Not sure about this. Stubtest would emit an error here if it encountered this, since `__all__` is different in the stub to what it is at runtime. Rather than trying to synthesise a "corrected version of `__all__`", I think stubgen should just omit it from the generated stub in this situation

Reply:
Omitting it also feels wrong though, and should generate a stubtest error (not sure if it currently does). Maybe the best option is to just emit `__all__` as it is at runtime, even if it includes some undefined names.

Reply:
Stubtest doesn't complain if `__all__` is omitted at runtime, no (though I think it _should_ -- see #13300).

We're in a bit of a catch-22 here if the runtime `__all__` has members that don't exist at runtime. In terms of typeshed's CI:
1. If `__all__` is different in the stub to what it is at runtime, then stubtest will complain. 
2. If `__all__` in the stub contains members that don't exist in the stub, pyright will complain.
3. If `__all__` is present at runtime but omitted in the stub, then neither will complain but, as you say, maybe stubtest _should_ complain...

I think you're right that emitting `__all__` as it is at runtime is probably the least opinionated thing to do. Maybe we should just add a comment in the test cases to say that we know this is weird, but there's no obviously correct solution for this situation

Reply:
Agree, and both stubtest and pyright are correct in those errors. Ultimately if `__all__` contains a name that doesn't exist, that's a bug in the runtime library, and it seems right that to work around it in typeshed we'll have to ignore an error (either through the stubtest allowlist or a type ignore).